Crash when TURN server requires credentials

### Your environment.
  - Version: cbd86cd4c700c72ceef2b0848d89dd3f56cf604d
  - Browser: Firefox 75.0b11 (64-bit) Linux and Chromium 80.0.3987.162 

### What did you do?
Joining a conference with two peers. 

### What did you expect?
WebRTC connection to be over TURN

### What happened?
The turn server requires credentials which I couldn't figure out where to specify in the sfu.toml if they are not specified sfu just crashes with
```
2020-04-03 18:47:17.772 ERR NewWebRTCTransport api.NewPeerConnection InvalidAccessError: turn server credentials required
2020-04-03 18:47:17.772 INF rtc.GetOrNewRouter id=8e78b081-ad44-4aea-ac0e-e321eff2abc3#TOUYWG
2020-04-03 18:47:17.773 INF rtc.GetRouter id=8e78b081-ad44-4aea-ac0e-e321eff2abc3#TOUYWG
2020-04-03 18:47:17.773 INF rtc.AddRouter id=8e78b081-ad44-4aea-ac0e-e321eff2abc3#TOUYWG
2020-04-03 18:47:17.774 INF NewRouter id=8e78b081-ad44-4aea-ac0e-e321eff2abc3#TOUYWG
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe0 pc=0xcc775a]

goroutine 34 [running]:
github.com/pion/ion/pkg/rtc/transport.(*WebRTCTransport).Answer(0x0, 0x1, 0xc000102700, 0x6c6, 0x0, 0xc00038fa20, 0x2b, 0xc000161c00, 0x7f5a04ddd008, 0x0, ...)
	/app/code/pkg/rtc/transport/webrtctransport.go:266 +0x7a
github.com/pion/ion/pkg/node/sfu.publish(0xc00032f6b0, 0x12, 0xc0000e5d90)
	/app/code/pkg/node/sfu/internal.go:88 +0x488
github.com/pion/ion/pkg/node/sfu.handleRequest.func1(0xc00032f680, 0xc00032f740, 0xc00032f770)
	/app/code/pkg/node/sfu/internal.go:31 +0x2f2
github.com/cloudwebrtc/nats-protoo.(*NatsProtoo).handleRequest(0xc00018dc80, 0xc00032f680, 0xc0003379e0, 0x14, 0xc000337a00, 0x19)
	/root/go/pkg/mod/github.com/cloudwebrtc/nats-protoo@v0.0.0-20200328144814-d3c1c848d442/protoo.go:150 +0x1d4
github.com/cloudwebrtc/nats-protoo.(*NatsProtoo).handleMessage(0xc00018dc80, 0xc0000b8900, 0x820, 0x820, 0xc0003379e0, 0x14, 0xc000337a00, 0x19)
	/root/go/pkg/mod/github.com/cloudwebrtc/nats-protoo@v0.0.0-20200328144814-d3c1c848d442/protoo.go:105 +0x156
github.com/cloudwebrtc/nats-protoo.(*NatsProtoo).onRequest(0xc00018dc80, 0xc00036e0a0)
	/root/go/pkg/mod/github.com/cloudwebrtc/nats-protoo@v0.0.0-20200328144814-d3c1c848d442/protoo.go:96 +0x198
github.com/nats-io/nats%2ego.(*Conn).waitForMsgs(0xc0000c7b80, 0xc000184240)
	/root/go/pkg/mod/github.com/nats-io/nats.go@v1.9.1/nats.go:2147 +0x330
created by github.com/nats-io/nats%2ego.(*Conn).subscribe
	/root/go/pkg/mod/github.com/nats-io/nats.go@v1.9.1/nats.go:2987 +0x50e
```

The sfu.toml section looks like this:
```
[webrtc]
ice = ["stun:my.nebulon.space:3478", "turn:my.nebulon.space:3478"]
```
Is there a way to specify credentials within that uri schema?
And generally it should maybe nor crash if the server does require credentials.

Thanks!

--- a/cmd/sfu/main.go
+++ b/cmd/sfu/main.go
@@ -15,18 +15,22 @@ import (
 func init() {
 	var icePortStart, icePortEnd uint16
 
-	if len(conf.WebRTC.Ice.ICEPortRange) == 2 {
-		icePortStart = conf.WebRTC.Ice.ICEPortRange[0]
-		icePortEnd = conf.WebRTC.Ice.ICEPortRange[1]
+	if len(conf.WebRTC.ICEPortRange) == 2 {
+		icePortStart = conf.WebRTC.ICEPortRange[0]
+		icePortEnd = conf.WebRTC.ICEPortRange[1]
 	}
 
 	log.Init(conf.Log.Level)
-	ice := webrtc.ICEServer{
-		URLs:       conf.WebRTC.Ice.URLs,
-		Username:   conf.WebRTC.Ice.Username,
-		Credential: conf.WebRTC.Ice.Credential,
+	var iceServers []webrtc.ICEServer
+	for _, iceServer := range conf.WebRTC.ICEServers {
+		s := webrtc.ICEServer{
+			URLs:       iceServer.URLs,
+			Username:   iceServer.Username,
+			Credential: iceServer.Credential,
+		}
+		iceServers = append(iceServers, s)
 	}
-	if err := rtc.Init(conf.Rtp.Port, ice, icePortStart, icePortEnd, "", ""); err != nil {
+	if err := rtc.Init(conf.Rtp.Port, iceServers, icePortStart, icePortEnd, "", ""); err != nil {
 		panic(err)
 	}
 }

--- a/configs/sfu.toml
+++ b/configs/sfu.toml
@@ -8,15 +8,17 @@ dc = "dc1"
 addr = "127.0.0.1"
 
 [webrtc]
-
-# if ion behind nat, uncomment webrtc.ice configs
-[webrtc.ice]
-# urls = ["stun:stun.stunprotocol.org:3478"]
-# username = ""
-# credential = ""
-
 # Range of ports that ion accepts WebRTC traffic on
-ephemeral-udp-port-range = []
+# portrange = []
+
+# if sfu behind nat, set iceserver
+# [[webrtc.iceserver]]
+# urls = ["stun:stun.stunprotocol.org:3478"]
+
+# [[webrtc.iceserver]]
+# urls = ["turn:turn.awsome.org:3478"]
+# username = "awsome"
+# credential = "awsome"
 
 [rtp]
 # listen port

--- a/pkg/conf/sfu/conf.go
+++ b/pkg/conf/sfu/conf.go
@@ -44,15 +44,15 @@ type nats struct {
 	URL string `mapstructure:"url"`
 }
 
-type ICE struct {
-	URLs         []string `mapstructure:"urls"`
-	Username     string   `mapstructure:"username"`
-	Credential   string   `mapstructure:"credential"`
-	ICEPortRange []uint16 `mapstructure:"ephemeral-udp-port-range"`
+type iceserver struct {
+	URLs       []string `mapstructure:"urls"`
+	Username   string   `mapstructure:"username"`
+	Credential string   `mapstructure:"credential"`
 }
 
 type webrtc struct {
-	Ice ICE `mapstructure:"ice"`
+	ICEPortRange []uint16    `mapstructure:"portrange"`
+	ICEServers   []iceserver `mapstructure:"iceserver"`
 }
 
 type rtp struct {

--- a/pkg/rtc/rtc.go
+++ b/pkg/rtc/rtc.go
@@ -28,10 +28,10 @@ var (
 )
 
 // Init port and ice urls
-func Init(port int, ice webrtc.ICEServer, icePortStart, icePortEnd uint16, kcpKey, kcpSalt string) error {
+func Init(port int, iceServers []webrtc.ICEServer, icePortStart, icePortEnd uint16, kcpKey, kcpSalt string) error {
 
 	//init ice urls and ICE settings
-	if err := transport.InitWebRTC(ice, icePortStart, icePortEnd); err != nil {
+	if err := transport.InitWebRTC(iceServers, icePortStart, icePortEnd); err != nil {
 		return err
 	}
 

--- a/pkg/rtc/transport/webrtctransport.go
+++ b/pkg/rtc/transport/webrtctransport.go
@@ -33,13 +33,13 @@ var (
 )
 
 // InitWebRTC init WebRTCTransport setting
-func InitWebRTC(ice webrtc.ICEServer, icePortStart, icePortEnd uint16) error {
+func InitWebRTC(iceServers []webrtc.ICEServer, icePortStart, icePortEnd uint16) error {
 	var err error
 	if icePortStart != 0 || icePortEnd != 0 {
 		err = setting.SetEphemeralUDPPortRange(icePortStart, icePortEnd)
 	}
 
-	cfg.ICEServers = []webrtc.ICEServer{ice}
+	cfg.ICEServers = iceServers
 	return err
 }
 

